python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -250,12 +267,19 @@ def apply(self, catalog):
             (True means selected).
         """
         flagField = self.fluxField + "_flag"
-        if flagField in catalog.schema:
-            selected = np.logical_not(catalog[flagField])
+        if isinstance(catalog, DataFrame) or isinstance(catalog, Table):
+            if flagField in catalog.columns:
```

Comment:
Is there any way to restructure this so that we can take the `np.ones()` blocks out of both `if`s? If we set `schema = catalog.schema if isinstance(catalog, SimpleCatalog) else catalog.columns` at the start, we could do one `if flagField in schema` at outside the `isinstance` check and I think this whole block would then be simpler.

Alternately, a lambda or closure like `isFieldInCatalog(field)` that encapsulates the `schema` vs. `columns` split.

Reply:
I'll add a little `_` method, I guess it's used enough to be worthwhile.

Reply:
I am about to add the following to `utils.py` in `pipe_analysis`:
```
def getSchema(catalog):
    """Helper function to determine "schema" of catalog.

    This will be the list of columns if the catlaog type is a pandas DataFrame,
    or the schema object if it is an `lsst.afw.table.SourceCatalog`.
    """
    if isinstance(catalog, pd.DataFrame):
        schema = catalog.columns
    elif isinstance(catalog, pd.Series):
        schema = [catalog.name, ]
    else:
        schema = catalog.schema
    return schema
```

Reply:
If we are planning to use pandas more broadly, that shouldn't go into pipe_analysis, but rather some lower level LSST package.